### DISCUSSION TITLE
Uxarray Quality of Life Improvements

### DISCUSSION THREAD

**Discussion by anissa111** [UXARRAY/uxarray#88] [Ideas]
Hello! Following a discussion with @michaelavs and @erogluorhan, here is a discussion post for collaborator input on some quality of life improvements we've discussed.

Feel free to comment on or add your own!

**Comment by anissa111** [comment]
Regarding code refactoring, see #86 for some of @erogluorhan's suggestions and feel free to expand that on as you see fit.

**Comment by anissa111** [comment]
Removing unimplemented functions, as the draft API is now included in the repo.

**Comment by anissa111** [comment]
Improving testing coverage. See #52 for more proposed detail

**Comment by erogluorhan** [comment]
Uxarray [Usage examples](https://github.com/UXARRAY/uxarray/issues/69) issue is a good one for this!

**Comment by anissa111** [comment]
General documentation improvement, including examples and making sure the pages are rendering correctly as a condition for PR approval

**Comment by anissa111** [comment]
A reorganization of the `helpers.py` function.
- rename to `util.py`?
- pull out groups of connected functions that could have their own file, like the jacobian functions
-

**Comment by erogluorhan** [comment]
[Contributor's Guide](https://github.com/UXARRAY/uxarray/issues/21) is also one thing that needs to be put together soon!

**Comment by anissa111** [comment]
Creation of a better storage area for testing files, similar to `xarray.testing` or geocat-datafiles

**Comment by anissa111** [comment]
I would also really like a function like `Grid.__str__` that allows for formatted printing of class variables without needing to know the cf convention names to access

**Reply by rajeeja** [reply]
How about just using Grid.ds ?

**Reply by anissa111** [reply]
Yeah! I think something like that would definitely work. This is a relatively minor point anyway, but I think it'd be cool if we could have something (possibly just `Grid.ds`, possibly more) that prints out if we do a `print(ux_grid)` type call

**Reply by dcherian** [reply]
:+1: for `__repr__` and `_repr_html_` Nobody actually coded it up but here are ideas we discussed for xgcm: https://github.com/xgcm/xgcm/issues/276
